client/src/assets/js/app/typing-lesson.js

```diff
@@ -55,6 +56,7 @@
         this.curCombo = this.combos[this.comboIdx];
         this.curLetter = this.curCombo[this.letterIdx];
         this.combosLeft = this.combos.length - this.comboIdx;
+        this.lessonCompletionPercent = 100 - (this.combosLeft / this.combos.length * 100);
```

Comment:
`lessonCompletionPercent` is a tad redundant since it is a property of `lesson`.